src/libraries/System.IO.FileSystem.Watcher/src/System/IO/FileSystemEventArgs.cs

```diff
@@ -17,14 +17,18 @@ public class FileSystemEventArgs : EventArgs
         /// </devdoc>
         public FileSystemEventArgs(WatcherChangeTypes changeType, string directory, string? name)
         {
+            ArgumentNullException.ThrowIfNull(directory);
+
             _changeType = changeType;
             _name = name;
-            _fullPath = Path.Join(Path.GetFullPath(directory), name);
 
-            if (string.IsNullOrWhiteSpace(name))
+            _fullPath = Path.GetFullPath((string.IsNullOrWhiteSpace(directory), string.IsNullOrWhiteSpace(name)) switch
```

Comment:
Isn't a file name of all whitespace a valid name, at least on Unix?

Reply:
> not sure if it should be addressed right now.

It seems the previous change also broke this.

I'm not sure the previous change was fully thought through.  I'm wondering if it should just be reverted and then revisited if desired.

Reply:
I would prefer to keep the previous change and just address the whitespace filename issue, I don't think more problems will come-up from the previous change 🤞

Reply:
Thanks. I'll change to `IsNullOrEmpty`.